src/sentry/api/endpoints/organization_details.py

```diff
@@ -248,6 +288,55 @@ def validate(self, attrs):
                 )
         return attrs
 
+    def save_trusted_relays(self, incoming, changed_data, organization):
+        timestamp_now = datetime.utcnow().replace(tzinfo=UTC).isoformat()
+        option_key = "sentry:trusted-relays"
+        try:
+            # get what we already have
+            existing = (
+                OrganizationOption.objects.get(organization=organization, key=option_key) or {}
+            )
+            key_dict = {val.get("public_key"): val for val in existing.value}
```

Comment:
Can `existing.value` ever be `None`? If so, we should do this:

```suggestion
            key_dict = {val.get("public_key"): val for val in (existing.value or [])}
```